src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -65,62 +65,63 @@ public InitCodeView generateInitCode(
       Iterable<Field> fields,
       SymbolTable table,
       TestValueGenerator valueGenerator) {
-    Map<String, Object> initFieldStructure = createSampleInitFieldStructure(context);
-    InitCodeGeneratorContext initCodeContext =
-        InitCodeGeneratorContext.newBuilder()
-            .symbolTable(table)
-            .valueGenerator(valueGenerator)
-            .initStructure(initFieldStructure)
-            .initObjectName(Name.from("request"))
-            .initObjectType(context.getMethod().getInputType())
-            .flattenedFields(Lists.newArrayList(fields))
-            .build();
-    InitCodeGenerator generator = new InitCodeGenerator();
-    InitCode initCode = generator.generate(initCodeContext);
-
-    return buildInitCodeView(context, initCode);
+    return buildInitCodeView(
+        context,
+        generateInitCodeHelper(
+            Lists.newArrayList(fields),
+            table,
+            valueGenerator,
+            createSampleInitFieldStructure(context),
+            Name.from("request"),
+            context.getMethod().getInputType()));
```

Comment:
I will rethink whether there is a clearer way to provide helpers, otherwise I will revert it.